test/signing_test.go

```diff
@@ -8,15 +8,436 @@ import (
 
 	"github.com/Azure/dalec"
 	"github.com/Azure/dalec/test/testenv"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/client/llb"
 	gwclient "github.com/moby/buildkit/frontend/gateway/client"
-	"github.com/stretchr/testify/assert"
+	"gotest.tools/v3/assert"
 )
 
-func distroSigningTest(t *testing.T, spec *dalec.Spec, buildTarget string) testenv.TestFunc {
+func runTest(t *testing.T, f testenv.TestFunc, opts ...testenv.TestRunnerOpt) {
+	t.Helper()
+	ctx := startTestSpan(baseCtx, t)
+	testEnv.RunTest(ctx, t, f, opts...)
+}
+
+func newSimpleSpec() *dalec.Spec {
+	spec := fillMetadata("foo", &dalec.Spec{
+		Sources: map[string]dalec.Source{
+			"foo": {
+				Inline: &dalec.SourceInline{
+					File: &dalec.SourceInlineFile{
+						Contents: "#!/usr/bin/env bash\necho \"hello, world!\"\n",
+					},
+				},
+			},
+		},
+		Build: dalec.ArtifactBuild{
+			Steps: []dalec.BuildStep{
+				{
+					Command: "/bin/true",
+				},
+			},
+		},
+		Artifacts: dalec.Artifacts{
+			Binaries: map[string]dalec.ArtifactConfig{
+				"foo": {},
+			},
+		},
+	})
+
+	return spec
+}
+
+func linuxSigningTests(ctx context.Context, testConfig testLinuxConfig) func(*testing.T) {
+	return func(t *testing.T) {
+		newSigningSpec := func() *dalec.Spec {
+			spec := newSimpleSpec()
+			spec.PackageConfig = &dalec.PackageConfig{
+				Signer: &dalec.PackageSigner{
+					Frontend: &dalec.Frontend{
+						Image: phonySignerRef,
+					},
+				},
+			}
+
+			return spec
+		}
+
+		t.Run("root config", func(t *testing.T) {
+			t.Parallel()
+			spec := newSigningSpec()
+			runTest(t, distroSigningTest(t, spec, testConfig.SignTarget))
+		})
+
+		t.Run("with target config", func(t *testing.T) {
+			t.Parallel()
+			spec := newSigningSpec()
+			first, _, _ := strings.Cut(testConfig.SignTarget, "/")
+			spec.Targets = map[string]dalec.Target{
+				first: {
+					PackageConfig: &dalec.PackageConfig{
+						Signer: spec.PackageConfig.Signer,
+					},
+				},
+			}
+			spec.PackageConfig.Signer = nil
+
+			runTest(t, distroSigningTest(t, spec, testConfig.SignTarget))
+		})
+
+		t.Run("target config takes precedence when root config is there", func(t *testing.T) {
```

Comment:
This could use a similar check to see if the warning is provided.

Reply:
done -- had to implement the warning to get the test to pass :)